pkg/azurefile/azurefile.go

```diff
@@ -252,49 +252,37 @@ func GetFileShareInfo(id string) (string, string, string, string, error) {
 
 // check whether mountOptions contains file_mode, dir_mode, vers, if not, append default mode
 func appendDefaultMountOptions(mountOptions []string) []string {
-	fileModeFlag := false
-	dirModeFlag := false
-	versFlag := false
-	actimeoFlag := false
-	mfsymlinksFlag := false
-
-	for _, mountOption := range mountOptions {
-		if strings.HasPrefix(mountOption, fileMode) {
-			fileModeFlag = true
-		}
-		if strings.HasPrefix(mountOption, dirMode) {
-			dirModeFlag = true
-		}
-		if strings.HasPrefix(mountOption, vers) {
-			versFlag = true
-		}
-		if strings.HasPrefix(mountOption, actimeo) {
-			actimeoFlag = true
-		}
-		if strings.HasPrefix(mountOption, mfsymlinks) {
-			mfsymlinksFlag = true
-		}
+	var defaultMountOptions = map[string]string{
+		fileMode: defaultFileMode,
+		dirMode:  defaultDirMode,
+		vers:     defaultVers,
+		actimeo:  defaultActimeo,
 	}
 
-	allMountOptions := mountOptions
-	if !fileModeFlag {
-		allMountOptions = append(allMountOptions, fmt.Sprintf("%s=%s", fileMode, defaultFileMode))
-	}
+	// required for maintaing the order.
+	var keys = [5]string{fileMode, dirMode, vers, actimeo, mfsymlinks}
```

Comment:
why set `5` here? use a non-fixed array is better

Reply:
Fixed